### PR TITLE
[XLATensor] Add max_pool2d

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/max_pool2d.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+xla::Shape NodeOutputShape(const NodeOperand& input, int kernel_size,
+                           int stride, int padding) {
+  std::vector<xla::int64> stride_2d(2, stride);
+  std::vector<xla::int64> padding_2d(2, padding);
+  std::vector<xla::int64> kernel_size_2d(2, kernel_size);
+  auto lower_for_shape_fn =
+      [&stride_2d, &padding_2d,
+       &kernel_size_2d](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    XLA_CHECK(operands.size() == 1)
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/max_pool2d.cpp]: XLA_CHECK_EQ

File: torch_xla/csrc/helpers.h [hunk 2]
```diff
@@ -74,7 +75,7 @@ class XlaHelpers {
 
   // Creates an XLA padding configuration from a padding attribute value.
   static xla::PaddingConfig MakeXlaPaddingConfig(
-      const std::vector<int64_t>& padding);
+      tensorflow::gtl::ArraySlice<xla::int64> padding);
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/helpers.h]: I don't understand, can you elaborate?

[the remark this answers is not in the document]
Comment (dlibenzi) [reply torch_xla/csrc/helpers.h]: Check here:

https://github.com/abseil/abseil-cpp/blob/efccc502606bed768e50a6cd5806d8eb13e4e935/absl/types/span.h#L308

[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/helpers.h]: I still don't understand, `tensorflow::gtl::ArraySlice` is an alias for `absl::Span`.

[the remark this answers is not in the document]
Comment (dlibenzi) [reply torch_xla/csrc/helpers.h]: If you have a Span<int> you cannot build a span out of {1,2,3} init-list.
If you have a Span<const int> you can.

[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/helpers.h]: Sounds like in both cases we have a `Span`, what is the difference?

[the remark this answers is not in the document]
Comment (dlibenzi) [reply torch_xla/csrc/helpers.h]: Sorry ... comment lost angular brackets :)
Let's try again:

```
void foo(Span<int> a);
foo({1,2,3}); // Error

void moo(Span<const int> a);
moo({1,2,3}); // OK
```

**Comment by asuhan** [comment]
Addressed comments, PTAL.

**Pull request merged**